File: lc/py/072.py
class Solution:
    def sortColors(self, nums: List[int]) -> None:
        RED, WHITE, BLUE = range(3)
        r, w, b = 0, 0, len(nums)-1
        while w <= b:
            if nums[w] == RED:
                nums[w], nums[r] = nums[r], nums[w]
                r += 1; w += 1
            elif nums[w] == BLUE:
                nums[w], nums[b] = nums[b], nums[w]
                b -= 1
            else:
                w += 1

File: lc/c/072.c
void swap(int *a, int *b) {
    int tmp = *a;
    *a = *b;
    *b = tmp;
}

void sortColors(int* nums, int numsSize){
    enum {RED, WHITE, BLUE};
    int r = 0, w = 0, b = numsSize-1;
    while (w <= b) {
        if (nums[w] == RED)
            swap(&nums[w++], &nums[r++]);
        else if (nums[w] == BLUE)
            swap(&nums[w], &nums[b--]);
        else
            w++;
    }
}
